nomad/structs/config/artifact.go

```diff
@@ -39,6 +39,11 @@ type ArtifactConfig struct {
 	// artifact downloader can write only to the task sandbox directory, and can
 	// read only from specific locations on the host filesystem.
 	DisableFilesystemIsolation *bool `hcl:"disable_filesystem_isolation"`
+
+	// SetEnvironmentVariables is a comma-separated list of environment
+	// variable names to inherit from the Nomad Client and set in the artifact
+	// download sandbox process.
+	SetEnvironmentVariables *string `hcl:"set_environment_variables"`
```

Comment:
On the name "set": is there future in which we might want to let users set specific env vars, rather than having them inherit from the client?

Reply:
I think we can continue using this field to support setting custom values, just by interpreting '=' as key+value,  e.g.

```
set_environment_variables = "BLAH,PATH=/opt/custom/bin"
```

But that would be beyond compatibility support, so that's a PR for another day